utils/src/modules/repository.rs

```diff
@@ -32,19 +36,43 @@ impl Repository {
     }
 
     pub fn update_at(&mut self, key: String, value: Bytes, activation_time: Option<u64>) {
-        let mode = Repository::resolve_update_mode(activation_time);
-        let (current, _) = self.storage.get(&key);
-        let new_value = match mode {
-            UpdateMode::SetFuture => (current, Some((value.clone(), activation_time.unwrap()))),
-            UpdateMode::ClearFuture => (current, None),
-            UpdateMode::Current => (value.clone(), None),
+        let now = get_block_time();
+        let value_for_event = value.clone();
+        let new_value: Record = match activation_time {
```

Comment:
It's really hard to read this `match` statement. I feel lost especially in the last complex `Some(activation_time)` arm. Btw. I wasn't aware of the Some if syntax. I liked the approach of resolving cases in a separate function. Then the main `update_at()` simply sets values. I know, comments describe each case, but I prefer simple functions, all the complex logic can be hidden.
What do think?

Reply:
Let's merge this one and you can propose your version.